src/StudyViewer/StudyDetails.jsx

```diff
@@ -298,6 +298,13 @@ class StudyDetails extends React.Component {
      // button. if there are more than 1 with different configs, TODO fix
      const requestAccessConfig = this.props.studyViewerConfig.buttons && this.props.studyViewerConfig.buttons.find((e) => e.type === 'request_access');
 
+     let loginAlertMessage = 'Researchers are required to log in to request access to this dataset. Use the button above to login or create an account.';
+     if (!(!this.props.data.requiredIdpField
+      || this.props.data.requiredIdpField === this.props.user?.idp)) {
```

Comment:
isn't this equivalent to  `if (this.props.data?.requiredIdpField && this.props.data.requiredIdpField === this.props.user?.idp)` ?

Reply:
thanks for pointing out i think its the same as 
`this.props.data?.requiredIdpField && this.props.data.requiredIdpField !== this.props.user?.idp`